### ISSUE TITLE
HTML Graphics plugin - configuration screen locked in “loading”

### ISSUE THREAD

**Issue by cschafranski** [gapitio/gapit-htmlgraphics-panel#32]
**What happened**: I'm using the HTML graphics plugin, at first the component enables the encoding option normally, after a few accesses the configuration screen is locked in "loading".

**What you expected to happen**: I expected the components to render correctly and the code option to be enabled.

**How to reproduce it (as minimally and precisely as possible)**: When entering the panel and entering the HTML graphical plugin configuration screen, the code inputs are not rendered and are "loading" infinitely.

**Anything else we need to know?**: I already cleared the page cache and the problem persisted.

**Environment**:

- Grafana version: **Grafana v8.0.3 (cae5c5e46b)**
- Plugin (gapit-htmlgraphics-panel) version: **1.3.3**
- Data source type & version: **InfluxDB 2.0**
- OS Grafana is installed on:  **CentOS Linux 8 / podman version 3.0.2-dev**
- User OS & Browser: **Chrome**
- Grafana plugins: **gapit-htmlgraphics-panel @ 1.3.3** , **grafana-clock-panel @ 1.1.3**
- Others: 
![image](https://user-images.githubusercontent.com/82821559/129190741-48739f2e-0b9a-411e-80bd-42bbf7b4cebd.png)

**Comment by cschafranski** [comment]
Hi @ZuperZee 

The problem is that the edit mode is locked with this "loading" shown in the image.

**Comment by ZuperZee** [comment]
Always?  Or is it after something is done?

**Comment by ZuperZee** [comment]
If you open the Network window (ctrl+shift+i -> Network) do you see these 
![image](https://user-images.githubusercontent.com/39485579/129197559-0e79df56-57d1-444d-9dcf-42213705b363.png)

And do you have any errors in the console (ctrl+shift+j)?

**Comment by cschafranski** [comment]
The error permanently happens after adding an svg to the edit panel:

![image](https://user-images.githubusercontent.com/82821559/129198520-d6680cfd-67d4-4025-b472-bb54e215a1fa.png)

When entering this screen, the console has some errors:
![image](https://user-images.githubusercontent.com/82821559/129198826-aff5af93-f0d4-400f-bb9c-9b1cd0601040.png)

Network window:

![image](https://user-images.githubusercontent.com/82821559/129198961-11acc563-c673-4af4-8987-c73b27a6e1d3.png)

**Comment by ZuperZee** [comment]
Hmm, interesting. Seems like it couldn't find the editor files.
Does it happen if you create a new panel or do you have to add an SVG to it and is it any SVG?

**Comment by cschafranski** [comment]
When trying to create a new panel using the same plugin, the problem persists and I can't add code or new svg.

![image](https://user-images.githubusercontent.com/82821559/129200924-c71bead4-e5d3-4374-8aa4-67de6929cfc4.png)

**Comment by cschafranski** [comment]
Network:

![image](https://user-images.githubusercontent.com/82821559/129201141-346ac6db-1659-4687-bd12-311f7dba62ff.png)

**Comment by ZuperZee** [comment]
A quick think you could try is to disable the cache and press ctrl+f5
![image](https://user-images.githubusercontent.com/39485579/129201706-acab2fd8-b6dc-4c69-b414-35ba87157ece.png)

**Comment by cschafranski** [comment]
loader.js look this:

![image](https://user-images.githubusercontent.com/82821559/129201883-0228309d-d7f7-45e3-afa8-e14e582e450a.png)

**Comment by ZuperZee** [comment]
Ah, yeah a full reload (disable cache) will most likely fix it

**Comment by cschafranski** [comment]
Even with the cache disabled the problem persists:
![image](https://user-images.githubusercontent.com/82821559/129202179-57053dc6-9cdc-4461-ab55-8ba2dcd6c5ba.png)

**Comment by ZuperZee** [comment]
Is the loader.js status  200?
![image](https://user-images.githubusercontent.com/39485579/129202422-e844a38e-8011-44b0-8f82-8a94b58e6b5a.png)

**Comment by cschafranski** [comment]
Now it has status 200, previously it was status 304

![image](https://user-images.githubusercontent.com/82821559/129202686-7ec68f6a-9dcf-4c15-817b-13538a3a7208.png)

**Comment by ZuperZee** [comment]
I can see there are 2 loader.js files transferred. What does the one below it say?

**Comment by cschafranski** [comment]
![image](https://user-images.githubusercontent.com/82821559/129202974-bfbf80a8-7a0a-434e-aa2b-0382caa6ae45.png)

**Comment by ZuperZee** [comment]
And editor.main.js?
![image](https://user-images.githubusercontent.com/39485579/129203175-42c14955-9369-491a-b2e4-2cc5ed04c599.png)

**Comment by ZuperZee** [comment]
Could you try importing this dashboard and see if the bug still happens.
```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "id": 20,
  "links": [],
  "panels": [
    {
      "datasource": "-- Grafana --",
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "SVGBaseFix": true,
        "add100Percentage": true,
        "centerAlignContent": true,
        "codeData": "{\n  \"text\": \"Random text\"\n}",
        "css": "* {\n  font-family: Open Sans;\n}\n\n.box {\n  border: solid #555 2px;\n  border-radius: 10px;\n  padding: 10px 20px;\n}\n",
        "dynamicData": false,
        "html": "<div style=\"text-align: center;\">\n  <div class=\"box\" id=\"htmlgraphics-text\"></div>\n  <br />\n  <div class=\"box\" id=\"htmlgraphics-value\"></div>\n</div>\n",
        "onInit": "// Sets the text from customProperties\nconst htmlgraphicsText = htmlNode.getElementById('htmlgraphics-text');\n\nif (htmlgraphicsText) {\n  htmlgraphicsText.textContent = customProperties.text;\n\n  // Change the text color based on the theme\n  if (theme.isDark) {\n    htmlgraphicsText.style.color = 'green';\n  } else {\n    htmlgraphicsText.style.color = 'red';\n  }\n}\n",
        "onRender": "// Sets the value from the first series on every refresh\nconst htmlgraphicsValue = htmlNode.getElementById('htmlgraphics-value');\n\nif (htmlgraphicsValue) {\n  htmlgraphicsValue.textContent = data.series[0].fields[1].state.calcs.last;\n}\n",
        "overflow": "visible",
        "panelupdateOnMount": true,
        "renderOnMount": true
      },
      "targets": [
        {
          "queryType": "randomWalk",
          "refId": "A"
        }
      ],
      "title": "Panel Title",
      "type": "gapit-htmlgraphics-panel"
    }
  ],
  "schemaVersion": 30,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "Loading bug",
  "uid": "0CM_j6Mnz",
  "version": 1
}
```

![image](https://user-images.githubusercontent.com/39485579/129203859-df61fdfc-f126-40f5-9356-2cd6f21b998b.png)

**Comment by cschafranski** [comment]
The bug still happens.

![image](https://user-images.githubusercontent.com/82821559/129205184-54ba24a3-7698-4bc0-9823-8be2f3d57cbe.png)

**Comment by ZuperZee** [comment]
Ok, I'm gonna look at this a little more tomorrow. Could you try to provide some more information about how this is hosted? Docker, are you using a VPN or SSH to a server ETC. This is most likely a network issue causing the loader.js to not find the files.
You could try to reinstall the plugin and/or restart the grafana server.

**Comment by ZuperZee** [comment]
I'm having difficulties recreating the issue you're getting.

Could your try providing some information about how the Grafana hosted, how the plugin was installed, how you're accessing it and/or any information that could help recreate the issue?

**Comment by cschafranski** [comment]
Environment:

Grafana version: Grafana v8.0.3 (cae5c5e46b)
Plugin (gapit-htmlgraphics-panel) version: 1.3.3
Data source type & version: InfluxDB 2.0
OS Grafana is installed on: CentOS Linux 8 / podman version 3.0.2-dev
User OS & Browser: Chrome
Grafana plugins: gapit-htmlgraphics-panel @ 1.3.3 , grafana-clock-panel @ 1.1.3

**Comment by ZuperZee** [comment]
1. **How did you install the plugin?** Was it with `grafana-cli plugins install gapit-htmlgraphics-panel`?
2. **How are you accessing the Grafana?** Is it installed locally on your computer or is it through a proxy or ssh?
3. **How is the Grafana configured?** Default or did you change any config parameters?

**Comment by cschafranski** [comment]
1. The installation was done via CLI, reference: https://grafana.com/grafana/plugins/gapit-htmlgraphics-panel/
2. Through a proxy on the IBM platform on a vpc
3. Default

**Comment by ZuperZee** [comment]
Does the issue persist on other browsers?

**Comment by cschafranski** [comment]
Yes, can I create a user for you to access the application?

I just need you to give me an email so that I can send you the login and password.

**Comment by ZuperZee** [comment]
That would be great. 
My work mail: oklk@gapit.io

**Comment by cschafranski** [comment]
See if you got an invite.

**Comment by ZuperZee** [comment]
Doesn't seem like I got an invite.

**Comment by cschafranski** [comment]
I sent a new email with login and password

**Comment by ZuperZee** [comment]
Thx, got it. I'll look at it tomorrow.

**Comment by ZuperZee** [comment]
After logging in, I'm able to see the editor
![image](https://user-images.githubusercontent.com/39485579/129947420-167cabba-fd0a-4da0-b422-9ebea1ca52f4.png)

**Comment by ZuperZee** [comment]
Seems like there is an issue when the influxdb flux editor is loaded at the same time as the htmlgraphics editor.

**Comment by cschafranski** [comment]
Can you simulate the problem now? is it possible to fix?

**Comment by ZuperZee** [comment]
Loading both the influx editor and htmlgraphics editor causes the editors to stop working.
Loading one (only tried htmlgraphics) of the editors and then load both, both will load properly.
![loading-bug](https://user-images.githubusercontent.com/39485579/130056788-67c02c6f-83f1-4e05-827d-f3b9ae851cc9.gif)

**Comment by ZuperZee** [comment]
I'll try to fix it. Will probably be fixed in the next release (~2 weeks)

**Comment by ZuperZee** [comment]
https://github.com/grafana/grafana/discussions/38430

**Comment by ZuperZee** [comment]
https://github.com/grafana/grafana/pull/38703

**Comment by cschafranski** [comment]
Hi, has the fix already been released?

**Comment by ZuperZee** [comment]
No, I made a PR to Grafana, which will make it possible to implement it smoothly grafana/grafana#38703

**Comment by ZuperZee** [comment]
I've made a few solutions, but there are downsides to all of them. The PR will make it so there are no downsides.

**Comment by breinera** [comment]
I was wondering if there was a work-around for this issue until the 1.4.0 release is ready? I noticed the gif above seemed to show the editors loading but I can't quite determine what you are clicking on and the order in which you are clicking to make it work. Thanks

**Comment by neshorg** [comment]
@breinera The trick is to have a 2nd panel without the use of an "influxDB" datasource (as in his example GIF he's using -grafana- datasource in the *left* panel). 
- You'd then have to refresh te whole browser page in order to "initialize" this HTML Graphics plugin in this separate "dummy" panel (with the use of any other datasource except for InfluxDB). 
- Now the plugin wil load immediately because of this step. 
- After this you can go back and exit this panel 
- Return to your 'real' actual panel with InfluxDB & HTML Graphics plugin and everything will load nicely. (in his example GIF the *right* panel) 

In short, when first entering your dashboard to edit your actual panel, try to "edit" a separate different "dummy" panel with this plugin before going to your own actual panel.
